spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/configuration/PropertyUtil.java

```diff
@@ -65,6 +67,24 @@ public static <T> T getOrLegacyOrDefault(
     return property;
   }
 
+  /**
+   * Returns the property in the given relevance: property, defaultProperty
+   *
+   * @param propertyName the name of the property, used for logging
+   * @param propertySupplier a function to supply the property, may throw
+   * @param defaultProperty the default to apply if nothing else suits, may be null
+   * @param configCache the cache to save the property to, may be null
+   * @return the property resolved
+   * @param <T> the type of the property
+   */
+  public static <T> T getOrOrDefault(
+      final String propertyName,
+      final Supplier<T> propertySupplier,
+      final T defaultProperty,
+      final Map<String, Object> configCache) {
+    return getOrLegacyOrDefault(propertyName, propertySupplier, null, defaultProperty, configCache);
```

Comment:
🔧 Rather than providing `null` and requiring the additional `null` check in the `getOrLegacyOrDefault` method, you could pass a null supplier `() -> null`. You could then also move that to a constant and naming it something like `NO_LEGACY_PROPERTY`:

```java
return getOrLegacyOrDefault(propertyName, propertySupplier, NO_LEGACY_PROPERTY, defaultProperty, configCache);
```

Reply:
I like it! 😍

Reply:
Static constant won't work though because of the generic T, but will create a static method for that.